source/PlanetPanel.cpp

```diff
@@ -80,9 +80,9 @@ void PlanetPanel::Step()
 	}
 
 	// If the player starts a new game, exits the shipyard without buying
-	// anything, clicks to the bank, then returns to the shipyard and buys a
-	// ship, make sure they are shown an intro mission.
-	if(GetUI()->IsTop(this) || GetUI()->IsTop(bank.get()))
+	// anything, clicks to another location, then returns to the shipyard
+	// and buys a ship, make sure they are shown an intro mission.
+	if(GetUI()->IsTop(selectedPanel && selectedPanel != spaceport.get() ? selectedPanel : this))
```

Comment:
```suggestion
	if(GetUI()->IsTop(selectedPanel && selectedPanel != spaceport.get() ? selectedPanel : this))
```
can you maybe make something of this a variable, maybe it's just me but this looks kinda confusing

Reply:
yes very much, thank you